packages/grid/_modules_/grid/components/panel/filterPanel/GridFilterInputValue.tsx

```diff
@@ -44,16 +44,23 @@ export function GridFilterInputValue(props: GridTypeFilterInputValueProps & Text
 
   const onFilterChange = React.useCallback(
     (event) => {
+      let value = event.target.value;
+      if (type === 'singleSelect') {
```

Comment:
It's not obvious why we do this. Maybe would make it clearer?

```suggestion
      // NativeSelect only cast the value to a string. 
      if (type === 'singleSelect') {
```